### ISSUE TITLE
GraphQL query view

### ISSUE THREAD

**Issue by homburg** [mitmproxy/mitmproxy#4030]
#### Is your feature request related to a problem? Please describe.
GraphQL request query bodies are technically just json, but the query field is very hard to read with escaped newlines.

#### Describe the solution you'd like
I'd like to inspect a GraphQL request directly in mitmweb + mitmproxy with a readable query presentation

#### Additional context
I'm currently using my own [custom content-view](https://github.com/homburg/mitmproxy/blob/feature/graphql-content-view/mitmproxy/contentviews/graphql.py). I print the query field with expanded newlines below the rest of the body as json.

Raw body
```
{
    "operationName": null,
    "query": "{\n  stationWithEvaId(evaId: 8000105) {\n    name\n    location {\n      latitude\n      longitude\n    }\n    picture {\n      url\n    }\n  }\n}\n",
    "variables": null
}
```

GraphQL content-view:
```
{
  "variables": null,
  "operationName": null
}
---
{
  stationWithEvaId(evaId: 8000105) {
    name
    location {
      latitude
      longitude
    }
    picture {
      url
    }
  }
}
```

I'd like the graphql content-view to be auto-selected, but there's no graphql-specific headers to match on.

**Comment by mhils** [comment]
This sounds reasonable. I would be happy to merge a PR that employs some heuristics to detect GraphQL (https://graphql.org/learn/serving-over-http/#post-request), and then changes the formatting accordingly! :smiley:

**Comment by homburg** [comment]
Fixed by #4396
